experimental/adaptive-dialog/docs/recognizers-rules-steps-reference.md

```diff
@@ -0,0 +1,739 @@
+        Property = "dialog.addTodo.title"
+    },
+    // Add the current todo to the todo's list for this user.
+    new EditArray(EditArray.ArrayChangeType.Push, "user.todos", "dialog.addTodo.title"),
+    new SendActivity("Ok, I have added {dialog.addTodo.title} to your todos."),
+    new IfCondition()
+    {
+        Condition = new ExpressionEngine().Parse("toLower(dialog.addTodo.title) == 'call santa'"),
+        Steps = new List<IDialog>()
+        {
+            new SendActivity("Yes master. On it right now [You have unlocked an easter egg] :)")
+        }
+    },
+    new SendActivity("You now have {count(user.todos)} items in your todo.")
+}));
+```
+
+### SwitchCondition
+Used to represent branching in conversational flow based on the outcome of an expression evaluation. See [here][5] to learn more about the common expression language.
+
+``` C#
+// Create an adaptive dialog.
+var cardDialog = new AdaptiveDialog("cardDialog");
+cardDialog.AddRule(new IntentRule("ShowCards",
+    steps: new List<IDialog>() {
+        // Add choice input.
+        new ChoiceInput()
+        {
+            // Output from the user is automatically set to this property
+            Property = "turn.cardDialog.cardChoice",
+            // List of possible styles supported by choice prompt. 
+            Style = ListStyle.Auto,
+            Prompt = new ActivityTemplate("What card would you like to see?"),
+            Choices = new List<Choice>()
+            {
+                new Choice("Adaptive card"),
+                new Choice("Hero card"),
+                new Choice("Video card")
+            }
+        }, 
+        // Use SwitchCondition step to dispatch to right dialog based on choice input.
+        new SwitchCondition()
+        {
+            Condition = "turn.cardDialog.cardChoice",
+            Cases = new Dictionary<string, List<IDialog>>()
+            {
+                { "Adaptive card", new List<IDialog>() { new BeginDialog("ShowAdaptiveCardDialog")} },
+                { "Hero card", new List<IDialog>() { new BeginDialog("ShowHeroCardDialog")} },
+                { "Video card", new List<IDialog>() { new BeginDialog("ShowVideoCardDialog")} }
+            }
+        }
+}));
+```
+
+### EndTurn
+Ends the current turn of conversation.
+
+``` C#
+new EndTurn()
+```
+
+### BeginDialog
+Invoke and begin a new dialog. Begin dialog takes the target dialog's name and the target dialog can be any type of dialog including Adaptive dialog or Waterfall dialog etc.
+
+``` C#
+new BeginDialog("BookFlightDialog")
+```
+
+
+### EndDialog
+Ends the active dialog. 
+
+**Note:** Adaptive dialogs by default will end automatically if the dialog has run out of steps to execute. To override this behavior, simply set the `AutoEndDialog` property on Adaptive Dialog to false. 
+
+``` C#
+new EndDialog()
+```
+
+### CancelAllDialog
+Cancels all active dialogs including any active parent dialogs.
```

Comment:
what is the effect of this call? what happens with the next utterance? why would one use this vs. EndDialog?